controllers/component/component_adapter.go

```diff
@@ -105,6 +105,7 @@ func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, erro
 			return controller.RequeueWithError(err)
 		}
 	}
+	a.component, err = a.loader.GetComponent(a.client, a.context, a.component.Name, a.component.Namespace)
```

Comment:
I think we'll need to check the `err` after this call to see if we managed to get the component successfully.